java-checks-test-sources/src/main/java/checks/SwitchRedundantKeywordCheck.java

```diff
@@ -42,6 +42,29 @@ int switchExpression(String mode) {
 
   int switchStatement(String mode) {
     int result = 0;
+
+    switch (mode) {
+      case "a" -> { // Noncompliant
+        result = 1;
+      }
+      case "a1" -> { // compliant, two statements in block
+        mode = "";
+        result = 1;
+      }
+      case "b" -> {
+        if (mode.equals("b")) {
+          doSomethingElse();
+        }
+      }
+      case "c" -> { // compliant, return is not an expression
+        return 1;
+      }
+      case "d" -> {
+        throw new IllegalArgumentException();
```

Comment:
This is a false negative. We should report an issue because the throw statement does not need to be in a block.